tests/models/fhirsearch_perform_iter_test.py

```diff
@@ -0,0 +1,224 @@
+import io
+import json
+import os
+import unittest
+from unittest.mock import MagicMock, patch
+
+import fhirclient
+
+from fhirclient import server
+
+from fhirclient.models import bundle
+from fhirclient.models.fhirsearch import FHIRSearch
+from fhirclient.models.bundle import Bundle, BundleEntry
+
+
+class TestFHIRSearchIter(unittest.TestCase):
+    def setUp(self):
+        # self.mock_server = MockServer(tmpdir=os.path.join(os.path.dirname(__file__), '..', 'data', 'examples'))
+        self.mock_server = MagicMock()
+        self.search = FHIRSearch(resource_type=Bundle)
+        self.mock_bundle = self.instantiate_from('bundle-example.json')
+        self.mock_next_bundle = self.instantiate_from('bundle-example-page2.json')
+
+    def instantiate_from(self, filename):
+        # Construct the path to the example file
+        datadir = os.path.join(os.path.dirname(__file__), '..', 'data', 'examples')
+        filepath = os.path.join(datadir, filename)
+
+        # Read the JSON file and return a Bundle instance
+        with io.open(filepath, 'r', encoding='utf-8') as handle:
+            js = json.load(handle)
+
+        self.assertEqual("Bundle", js["resourceType"], "The resourceType is not 'Bundle'.")
+        return Bundle(js)
+
+    @patch('fhirclient._utils._fetch_next_page')
+    @patch('fhirclient._utils.iter_pages')
+    @patch('fhirclient.models.fhirsearch.FHIRSearch.perform')
+    def test_perform_iter_single_bundle(self, mock_perform, mock_iter_pages, mock_fetch_next_page):
+        self.mock_bundle = self.instantiate_from('bundle-example-page2.json')
+        mock_perform.return_value = self.mock_bundle
+        mock_iter_pages.side_effect = iter([self.mock_bundle])
+        mock_fetch_next_page.side_effect = [None]
```

Comment:
Your tests are fine - but I will say that all of this mocking can end up being brittle. Like here, you are using knowledge of the internals of `perform_iter` (that it calls `perform()`) and then mocking out a lot of the detailed utility code - where there might be bugs! But it's generally easy to get lost in the sauce when doing all this mocking. If another dev breaks this test in six month, they might not have the understanding of which micro-targeted line of code you are meant to be testing.

I know that it's a bit philosophical territory about how small unit tests vs functional tests vs whatever. But just practically, it can be tough to mock all the little interactions like this.

This is entirely optional for this PR, but if I were writing tests for this, I might:
- Add a testing dependency on the [responses](https://pypi.org/project/responses/) mocking library. Like in pyproject.toml's optional-dependencies section, add `responses` to the `tests` list of dependencies.
- Then in tests like this, you can mock things at the network boundary:
```
@responses.activate
def test_perform_iter_single_bundle(self):
    responses.add(responses.GET, "https://example.com/bundle-url/", json={bundle content...})
```

But again, what you have here is fine - just it can be annoying when writing tests to do all the mocking like this, trying to avoid the network. But `responses` lets you mock further back.